File: examples/example_hoeffding_tree.py
from skmultiflow.data.sine_generator import SineGenerator
from skmultiflow.trees.hoeffding_tree import HoeffdingTree
import time

def get_classifier_accuracy_test(c, a):
    start = time.perf_counter()
    for i in range(1000000):
        X,y = a.next_sample()
        c.partial_fit(X, y)

    right = 0
    wrong = 0
    for i in range(100000):
        X,y = a.next_sample()
        label = c.predict(X)
        if label == y[0]:
            right += 1
        else:
            wrong += 1
    acc = right / (right + wrong)
    end = time.perf_counter()
    return acc, end-start

a = SineGenerator()
c = HoeffdingTree(split_criterion="gini", no_preprune=True, leaf_prediction="nb")
test_result = get_classifier_accuracy_test(c, a)
print(test_result)

File: examples/example_hoeffding_tree.jl
include("src\\ConceptDrift.jl")

function get_classifier_accuracy_test(c::ConceptDrift.HoeffdingTree, a::ConceptDrift.AbstractDataStream)
    for i in 1:1000000
        X,y = ConceptDrift.next_sample!(a, 1)
        ConceptDrift.partial_fit!(c, vec(X), y[1])
    end

    right = 0
    wrong = 0
    for i in 1:100000
        X,y = ConceptDrift.next_sample!(a, 1)
        label = ConceptDrift.predict(c, vec(X))
        if label == y[1]
            right += 1
        else
            wrong += 1
        end
    end

    acc = right / (right + wrong)
end

# a = ConceptDrift.SineGenerator(0)
a = ConceptDrift.RandomTreeGenerator(min_leaf_depth=5)
c = ConceptDrift.HoeffdingTree()
test_result = @timed get_classifier_accuracy_test(c, a)